### PR TITLE
Martijn

### PR DESCRIPTION
**Pull request by mwheymans**
Changed files so that pdf version of book can be made.  Made also adaptations in Chapter 4 and 6, not in text but only in fig and table nrs.  Made most text changes to Chapter 3.

### CODE REVIEW

File: 03-SingleMissingdataimputation.Rmd [hunk 1]
```diff
@@ -106,81 +96,84 @@ complete(imp_mean)
...
+Yoy do regression imputation in SPSS via the Missing Value Analysis procedures. There are two options for regression imputation, the Regression option and the EM (Expectation Maximization) option. The Regression option in SPSS has some flaws in the estimation of the regression parameters (see paper von Hippel 2004). Therefore, we use the EM algorithm. This algorithm is a likelihood-based procedure. This means that the most likely values of the regression coefficients are estimated given the data and subsequently used to impute the missing value. This EM procedure gives the same results as first performing a normal regression analysis in the dataset and subsequently estimate the missing values from the regression equation, after the missing values have been excluded. We will compare the EM and regression procedures below.
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd]: De eerste zin zou ik hier anders formuleren: 
Regression imputation in SPSS can be performed via the Missing Value Analysis procedure.

Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd]: De eerste zin zou ik hier anders formuleren: 
Regression imputation in SPSS can be performed via the Missing Value Analysis procedure.

File: 03-SingleMissingdataimputation.Rmd [hunk 2]
```diff
@@ -106,81 +96,84 @@ complete(imp_mean)
...
 
-```{r fig43, echo = FALSE, fig.cap="Transfer of the Tampascale and Pain variables to the Predicted and Predictor Variables windows.", out.width='90%', fig.align='center'}
+```{r fig3-11, echo = FALSE, fig.cap="Transfer of the Tampascale and Pain variables to the Predicted and Predictor Variables windows.", out.width='90%', fig.align='center'}
 knitr::include_graphics("images/fig3.11.png")
 ```
 
-Step 4
-Click on Continue -> EM and Choose for Normal in the Distribution group. Than thick Save completed data and give the dataset a name, for example “ImpTampa_EM” (Figure \@ref(fig:fig44)). 
+Step 3
+Click on Continue -> EM and Choose for Normal in the Distribution group. Than thick Save completed data and give the dataset a name, for example “ImpTampa_EM” (Figure \@ref(fig:fig3-12)).
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd:122]: > Continue -> EM
Select Normal in the Distribution grop and Save completed data ...

File: 03-SingleMissingdataimputation.Rmd [hunk 3]
```diff
@@ -106,81 +96,84 @@ complete(imp_mean)
...
 
-```{r fig44, echo = FALSE, fig.cap="Name of dataset to save the EM results in.", out.width='90%', fig.align='center'}
+```{r fig3-12, echo = FALSE, fig.cap="Name of dataset to save the EM results in.", out.width='90%', fig.align='center'}
 knitr::include_graphics("images/fig3.12.png")
 ```
 
-Step 5
-Then Click Continue -> OK. The new dataset “ImpTampa_EM” will open in a new window in SPSS. In this dataset the imputed data of the Tampascale Variable together with the original data is stored (Figure \@ref(fig:fig45), first 15 patients are shown). Adjust the Width and Decimals of the Tampa scale variable in the Variable View window to 4 and 3 respectively to get the same results as in Figure 3.13. 
+Step 4
+Click Continue -> OK. The new dataset “ImpTampa_EM” will open in a new window in SPSS. In this dataset the imputed data of the Tampascale Variable together with the original data is stored (Figure \@ref(fig:fig3-13), first 15 patients are shown). Adjust the Width and Decimals of the Tampa scale variable in the Variable View window to 4 and 3 respectively to get the same results as in Figure 3.13.
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd:129]: > Continue -> OK

The new dataset....

File: 03-SingleMissingdataimputation.Rmd [hunk 4]
```diff
@@ -106,81 +96,84 @@ complete(imp_mean)
...
+Be aware that SPSS uses as default only quantitative variables to impute the missing values with the EM algorithm. If you want to include categorical variables in the imputation model as auxiliary variables, you have to define them as scale variables.
+
+#### Normal Linear Regression
 
-To estimate the linear regression model:
-From the menus choose: Analyze -> Regression -> Linear
+We now compare the EM procedure with a two-step linear regression procedure. We first estimate the relationship between Pain and the Tampa scale variable in the dataset with linear regression after we have excluded the cases with missing values in the Tampa scale variable. Subsequently, we use the regression coefficients from this regression model to estimate the imputed values in the Tampa scale variable. Start by estimating the linear regression model with the Tampa scale variable as outcome variable (because we have to predict missing values in this variable) and use the Pain variable as the independent variable.
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd:139]: We **can** compare the EM procedure with a..

File: 03-SingleMissingdataimputation.Rmd [hunk 5]
```diff
@@ -106,81 +96,84 @@ complete(imp_mean)
...
+```{r tab3-3, echo = FALSE, fig.cap="Result of the linear regression analysis.", out.width='90%', fig.align='center'}
 knitr::include_graphics("images/table3.3.png")
 ```
 
 The linear regression model can be described as: 
 
 Tampascale=32.005+1.410 ×Pain
 
-When we use this regression model to impute the missing values in the Tampa scale variable and we compare them with the EM estimates, we see that the results are the same.
+Now impute the missing values in the Tampa scale variable and compare them with the EM estimates. You see that the results are the same.
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd:161]: When the missing values in the Tampa scale variable are imputed and compared with the EM estimates, the results are the same.

File: 03-SingleMissingdataimputation.Rmd [hunk 6]
```diff
@@ -106,81 +96,84 @@ complete(imp_mean)
...
+```{r fig3-16, echo = FALSE, fig.cap="Relationship between the Tampa scale and the Pain variable.", out.width='90%', fig.align='center'}
 knitr::include_graphics("images/fig3.16.png")
 ```
 
 The Tampa scale variable is located on the y-axis. The imputed values of the Tampa scale variable (red dots) are located on the regression line that was used to generate the imputed values. The green dots are the observed data values.
 
 ### Regression imputation in R
 
-We will now continue with the application of regression imputation in R. We use the same dataset as we did for regression imputation in SPSS above. Regression imputation can be done in R by using the R code 3.5 with as method “norm.predict”. Only the Pain variable will be used to predict the missing values in the Tampa scale variable.
+You can aplly regression imputation in R with as method “norm.predict” in the mice function. The Pain variable is used to predict the missing values in the Tampa scale variable.
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd:177]: Regression imputation in R can be **applied** with as...

File: 03-SingleMissingdataimputation.Rmd [hunk 7]
```diff
@@ -193,94 +186,93 @@ imp.regress$imp$Tampascale # Extract the imputed values
...
+With Stochastic regression models imputation uncertainty is accounted for by adding extra error variance to the predicted values that are generated by the linear regression model. Stochastic regression can be activated in SPSS via the Missing Value Analysis and the Regression Estimation option. However, the Regression Estimation option generates incorrect regression coefficient estimates (von Hippel, 2004) and will therefore not further discussed.
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd]: With Stochastic regression models imputation, uncertainty is accounted for by adding extra error variance to the predicted values that are generated by the linear regression model. Stochastic regression can be performed in SPSS via the Missing Value Analysis and the Regression Estimation option. However, the Regression Estimation option in SPSS generates incorrect regression coefficient estimates (von Hippel, 2004) and will therefore not further be discussed.

Comment (iriseekhout) [reply 03-SingleMissingdataimputation.Rmd]: Misschien anders formuleren ? Uitleggen hoe je dit kunt doen in SPSS zodat je wel goede schattingen krijgt? Je kunt toch wel random error schatten? En die aan de predicted values van regression imputation toevoegen? Ook misschien zelf nog even testen wat von Hippel zegt.. want hij is niet een super betrouwbare bron!

File: 03-SingleMissingdataimputation.Rmd [hunk 8]
```diff
@@ -193,94 +186,93 @@ imp.regress$imp$Tampascale # Extract the imputed values
...
 
 ### Stochastic regression imputation in R
 
-In R stochastic regression imputation can be done with the mice function. There are two versions available in the mice function to apply stochastic regression imputation, one that is non-Bayesian and one that is called Bayesian stochastic regression imputation.
-We start with non-Bayesian stochastic regression imputation. We will use only the Pain variable to predict the missing values in the Tampa scale variable. The function that mice uses for this procedure is the mice.impute.norm.nob function. We will use for the example the same dataset as above including 50 low back pain patients with missing values in the Tampa scale variable. 
-Stochastic regression imputation is applied in R code 3.7 by using the “norm.nob” method.
+You can apply stochastic regression imputation in R with the mice function. For this you use as method "norm.nob".
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd:198]: You can **perform** stochastic

File: 03-SingleMissingdataimputation.Rmd [hunk 9]
```diff
@@ -193,94 +186,93 @@ imp.regress$imp$Tampascale # Extract the imputed values
...
-Bayesian stochastic regression imputation can be applied in SPSS and R. We start with an explanation of Bayesian Stochastic regression imputation in SPSS, followed by an application in R.
+The difference between the non-Bayesian imputation procedure that you applied above and Bayesian imputation is that with the latter method extra variation is added to the imputed values via the regression coefficients. The idea is used that there is not one true (population) regression coefficient but that the regression coefficient itself follows a (probability) distribution. This is in contrast to the frequentist idea, which assume that there is one true population parameter. In this case the regression coefficient (is reflected by the confidence interval). In this book (and most statistial books) the freuentist approach is used For more information about the theory of Bayesian statistics we refer to the books of Box and Tiao (1992), Enders (2012) and Gelman et al. (2004).
```
Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd]: The difference between the non-Bayesian imputation procedure that is described above and Bayesian imputation is that with the latter method the extra variation is added via the regression coefficients, indead of directly to the imputed values. The Bayesian idea is that there is not one population regression coefficient, but that the parameter should be drawn from a distribution. The frequentist idea is that there is one true population regression coefficient, which has uncertainty reflected in the confidence interval around it. In this book we mostly built on the frequentist idea. For more information about the theory of Bayesian statistics we refer to the books of Box and Tiao (1992), Enders (2012) and Gelman et al. (2004).

Comment (iriseekhout) [inline 03-SingleMissingdataimputation.Rmd]: The difference between the non-Bayesian imputation procedure that is described above and Bayesian imputation is that with the latter method the extra variation is added via the regression coefficients, indead of directly to the imputed values. The Bayesian idea is that there is not one population regression coefficient, but that the parameter should be drawn from a distribution. The frequentist idea is that there is one true population regression coefficient, which has uncertainty reflected in the confidence interval around it. In this book we mostly built on the frequentist idea. For more information about the theory of Bayesian statistics we refer to the books of Box and Tiao (1992), Enders (2012) and Gelman et al. (2004).

Comment (iriseekhout) [reply 03-SingleMissingdataimputation.Rmd]: Ik vond dit stukje niet zo duidelijk. Heb een poging gedaan tot herschrijven, maar weet niet of het nu al perfect is.

File: 03-SingleMissingdataimputation.Rmd (hunk 6 above)
Comment (child-health-statistics) [inline 03-SingleMissingdataimputation.Rmd:177]: apply in de eerste regel.

**Review by child-health-statistics — CHANGES_REQUESTED** [review]

**Comment by mwheymans** [comment]
Closed

**Pull request closed without merging**